src/viking.js

```diff
@@ -1,14 +1,75 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.strength = strength;
+    this.health = health;
+  }
+
+  attack() {
+    return this.strength
+  }
+
+  receiveDamage(damage) {
+    this.health -= damage
+  }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor(name, vikingHealth, vikingStrength) {
+    super(vikingHealth, vikingStrength)
+    this.name = name
+  }
+
+  battleCry() {
+    return ("Odin Owns You All!")
+  }
+
+  receiveDamage(damage) {
+    this.health = this.health - damage
+    if (this.health > 0) {
+      return (`${this.name} has received ${this.health} points of damage`)
```

Comment:
Hi Thu, here the message is:
    `  return (`${this.name} has received ${damage} points of damage`)`
    damage instead of this.health